[BUG] flaky test: TestRefresherCancelContext

did not change any refreshtoken related code, but ut failed.

=== RUN   TestRefresherCancelContext
    token_refresher_test.go:122: 
        	Error Trace:	token_refresher_test.go:122
        	Error:      	Test timeout
        	Test:       	TestRefresherCancelContext
        	Messages:   	TestRefresherCancelContext
--- FAIL: TestRefresherCancelContext (1.00s)

after retry, it succeeded.

Here is the detailed info.
https://github.com/Azure/fleet/actions/runs/2753311659/attempts/1
https://github.com/Azure/fleet/actions/runs/2753311659/attempts/2

--- a/pkg/authtoken/token_refresher_test.go
+++ b/pkg/authtoken/token_refresher_test.go
@@ -6,6 +6,7 @@ package authtoken
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"testing"
 	"time"
@@ -118,7 +119,7 @@ func TestRefresherCancelContext(t *testing.T) {
 	expectedErr := context.Canceled
 	select {
 	case err := <-testChan:
-		if err == expectedErr {
+		if errors.Is(err, expectedErr) {
 			return
 		}
 		assert.Fail(t, fmt.Sprintf("got error: \"%s\", expected error: \"%s\"", err.Error(), expectedErr))

